Fix/larger emojis in other contexts

josemak25 opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/4114


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if the test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
   1. The emoji and text should appear vertically centered with respect to each other, such that the baseline of the emoji is below the plain text and the middle of the text vertically aligns with the middle of the emoji
   2. Draft a message (of any length) with only emoji the emoji should appear very large (27pt) in the composer.
   3. Go to Settings -> Profile and add emoji(s)[😎] to your last name or first name and save your profile. now open the settings RHP and look at your name, the emoji appear to be bigger in size than the text.
- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->
   1. The emoji and text should appear vertically centered with respect to each other, such that the baseline of the emoji is below the plain text and the middle of the text vertically aligns with the middle of the emoji
   2. Draft a message (of any length) with only emoji the emoji should appear very large (27pt) in the composer.
   3. Go to Settings -> Profile and add emoji(s)[😎] to your last name or first name and save your profile. now open the settings RHP and look at your name, the emoji appear to be bigger in size than the text.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Chrome Web</summary>

https://user-images.githubusercontent.com/33870502/220120140-d5943671-4816-48cd-9903-4b499880cdb5.mov

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Safari Web</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/33870502/220120214-f953ddb4-a1df-494d-a05f-098fa96175d9.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/33870502/220120634-a78792d9-27ec-4d9d-ad5c-8c95b7e2c8e3.mov
</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/33870502/220120341-43cb24bd-d9e8-4535-b357-30aec4cdb7da.mp4

</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/33870502/220121807-0ad21fb6-4e5a-45c8-907a-25bfa71fc54b.mov

</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/33870502/220120814-7283f403-0857-4ec3-83aa-1bcb0ab19658.mp4

</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/33870502/220120774-a75321f3-51e2-49ab-8535-175e1843e6b9.mov

</details>

srikarparsi committed:
refactor wallet transfer onyx error frontend

srikarparsi committed:
reset errors property to null

iwiznia committed:
Use push notification ID instead of accountID

iwiznia committed:
Remove private prefix

neil-marcellini committed:
Initial quintuple tap gesture on header view

neil-marcellini committed:
Move quintuple tap gesture into the screen wrapper

neil-marcellini committed:
Add a small centered modal type

neil-marcellini committed:
Use Onyx to open and close the test tools modal

neil-marcellini committed:
Blank test tools modal in the screen wrapper

neil-marcellini committed:
Create a test tools modal

neil-marcellini committed:
Open the test tools menu on 5 taps in dev only

neil-marcellini committed:
Only render the test tools modal on dev

neil-marcellini committed:
Fix the gesture onEnd callback

neil-marcellini committed:
Fix iOS by running callbacks on the JS thread

neil-marcellini committed:
Somewhat fix the iOS modal

roryabraham committed:
Install lottie-react-native

roryabraham committed:
Save draft state

roryabraham committed:
Add lottie asset

Beamanator committed:
Clear / show validate code error messages

marktoman committed:
Update expensify-common

Commit:
remove form wrapper

Commit:
remove disabling on load

Commit:
fix lint

rawalyogendra committed:
Fix: iOS autoscroll when `<input/>` is focused

situchan committed:
fix top emoji cut off in multiline input

mountiny committed:
Automatically pin the newly created admin chat

mountiny committed:
Fix style

neil-marcellini committed:
Min width for mobile test tool modal

neil-marcellini committed:
lint

neil-marcellini committed:
Add missing prop type

neil-marcellini committed:
Standardize on TestToolsModal name

neil-marcellini committed:
Remove extra top margin

tienifr committed:
fix: issue

fedirjh committed:
Use `updateAutomaticTimezone` instead of `setPersonalDetails`

fedirjh committed:
Remove unused deprecated method setPersonalDetails

fedirjh committed:
Remove unused imports

fedirjh committed:
Remove setPersonalDetails export

fedirjh committed:
Remove unused method mergeLocalPersonalDetails

fedirjh committed:
Fix updateAutomaticTimezone parameter

roryabraham committed:
Set up react-native-web-lottie

Prince-Mendiratta committed:
pass date as moment

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

Prince-Mendiratta committed:
set maximum and minimum possible date

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

Prince-Mendiratta committed:
remove moment() conversion for API calls

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

tjferriss committed:
Create Your-Expensify-Partner-Manager Article

New partner manager article

Prince-Mendiratta committed:
use date object

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

maddylewis committed:
Update Your-Expensify-Partner-Manager

iwiznia committed:
Move pusher subscribe to session

iwiznia committed:
Styles and tests

Beamanator committed:
Set up new contact methods route

Beamanator committed:
Add contact method copy

Beamanator committed:
Move LoginField to contacts dir

Beamanator committed:
New ContactMethods page

Beamanator committed:
Link to new contact methods page & simplify

Beamanator committed:
Reorder for correct animation direction

Beamanator committed:
add plural contact methods

Beamanator committed:
Move page to contacts dir

Beamanator committed:
Fix title and previous page

Beamanator committed:
Navigate to contact methods page after adding login

Beamanator committed:
Fix lint errors

roryabraham committed:
Add new JSON lottie animations

roryabraham committed:
Replace existing gifs with lottie animations

roryabraham committed:
Replace gif in ReimbursementAccountLoadingIndicator

roryabraham committed:
add back DeeplinkWrapper

roryabraham committed:
Remove unused imports

maddylewis committed:
Update Your-Expensify-Partner-Manager

maddylewis committed:
Update Your-Expensify-Partner-Manager

roryabraham committed:
Remove unused assets

tjferriss committed:
Rename Your-Expensify-Partner-Manager to Your-Expensify-Partner-Manager.md

ArekChr commented on src/pages/DetailsPage.js:
```diff
@@ -148,7 +149,9 @@ class DetailsPage extends React.PureComponent {
-                                        {isSMSLogin ? this.props.toLocalPhone(details.displayName) : details.displayName}
+                                        {isSMSLogin
+                                            ? this.props.toLocalPhone(details.displayName)
+                                            : renderTextWithEmoji(details.displayName, {...styles.emojiMessageText, ...styles.profileEmojiText})}
```
Passing props to the render function looks unusual if the render function is imported outside. Maybe it's better to make a function component something like <TextEmoji>? wdyt @josemak25 ?

josemak25 replied:
I think this a TextEmoji works better, I would make these change right away

josemak25 commented on src/libs/renderTextWithEmoji.js:
```diff
@@ -0,0 +1,19 @@
+        <Text key={char + index} style={[EmojiUtils.isEmoji(char) ? {...styles} : undefined]}>
+            {char}
+        </Text>
+    ))) : text;
```
Basically, this function is called inside a text component where it's been used, we only want to return a Text component when the emoji conditions are met, else returning a string would be rendered inside the parent component.

With your proposal unifying the component return type means we have to return an extra layer Text component even when it's not an emoji

Below is an example of what I mean even when it's not an emoji text, shouldn't we try to keep this in one layer when it's not an emoji text?

```javascript
 // parent component where renderTextWithEmoji is called
<Text>
     // child component returned by renderTextWithEmoji
     <Text> Hello </Text>
</Text>
```

ArekChr replied:
But if we return Text, we don't have to worry about that if it's wrapped in a text component.
Maybe the question is will this component be always rendered inside the parent Text?

josemak25 replied:
@ArekChr Yea in all references where it is used, they are all rendered inside a parent component, most of which are conditionally rendered

ArekChr commented on src/components/TextEmoji.js:
```diff
@@ -0,0 +1,44 @@
+
+const propTypes = {
+    /** The message text to render */
+    text: PropTypes.string.isRequired,
```
Let's change this prop to children

josemak25 replied:
Got you

ArekChr commented on src/components/TextEmoji.js:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    return _.map([...Str.htmlDecode(props.text)], ((char, index) => (
+        <Text key={char + index} style={EmojiUtils.isEmoji(char) ? propsStyle : undefined}>
```
```suggestion
        <Text key={`${char}_${index}`} style={EmojiUtils.isEmoji(char) ? propsStyle : undefined}>
```

parasharrajat reviewed (changes requested):
Please Add tests and QA tests & Offline Tests. QA tests are non-technical steps that a user can take to verify PR changes. 

![image](https://user-images.githubusercontent.com/24370807/218469407-4d3ec6ea-84b8-48b1-8650-46c12c5b8a03.png)


Please complete the PR author Checklist. It is mandatory

josemak25 commented on src/components/TextEmoji.js:
```diff
@@ -0,0 +1,44 @@
+};
+
+const TextEmoji = (props) => {
+    const propsStyle = StyleUtils.parseStyleAsArray(props.style);
```
resolved

Beamanator committed:
Add spanish translations for contact methods

Prince-Mendiratta committed:
remove unnecessary reassignment

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

iwiznia committed:
Fix tests

grgia committed:
Add footer to help site, (no images or social icons yet)

grgia committed:
Use existing breakpoints

grgia committed:
Fix medium breakpoints

grgia committed:
Add icons for docs site, add bottom logo and update bottom padding

grgia committed:
add vertical logo

tienifr committed:
fix: adding country code before validate the phone number

grgia committed:
Add social icons and fix margin of vertical logo

grgia committed:
Align logo against side

redstar504 committed:
Upgrade image-manipulator package to allow Android GIF avatars

tienifr committed:
fix: adding country code before validate the phone number

tienifr committed:
fix: adding country code before validate

PankajAS committed:
Add Search feature for Workspace members list

1. Added common isKeywordMatch function
2. Add more fields in filter condition: firstName, lastName
3. Combine all of these into one if statement
4. Use common function like clear()

1. Put this.state.searchValue.trim().toLowerCase() this out of loop
2. Space formate

UI Adjustments

Match "No results found" text with new chat text

Fix lint errors

1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

1. Fix lint changes
2. Add policyMemberList check for "no results found" text

Fix Space lint

1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

1. Remove updateSearchValue Comment
2. Update OnSearch name to updateSearchValue
3. Inline conditions for Filter list data

Remove clearMembersAndSearchValue method

1. Remove unnecessary safety condition
2.Remove unnecessary comment for this function
3. split into multiple lines so fix lint

Add singinkey

Revert "Add singinkey"

This reverts commit d8b3e83fda9fbb07a3bb865cc7fa5b255778787d.

remove extra line

Revert "remove extra line"

This reverts commit 022e15693390fae1a8e1eeb4c6c483c21e61cc2f.

Update src/pages/workspace/WorkspaceMembersPage.js

Co-authored-by: Miroslav Stevanovic <97473779+0xmiroslav@users.noreply.github.com>

Fix lint errors

1. Change to "Clear search input value and selected members"
2. change selectedEmployees to selectedMembers
3. Change to "Check if value matches with keyword"

1. Fix lint changes
2. Add policyMemberList check for "no results found" text

Fix Space lint

1. Add JSDoc for onSearch function
2. Remove commnet for Clear function
3. Change Clear method name to clearMembersAndSearchValue
4. Revert selectedMembers name with selectedEmployees

Remove clearMembersAndSearchValue method

1. Remove unnecessary safety condition
2.Remove unnecessary comment for this function
3. split into multiple lines so fix lint

Add singinkey

Revert "Add singinkey"

This reverts commit d8b3e83fda9fbb07a3bb865cc7fa5b255778787d.

remove extra line

Revert "remove extra line"

This reverts commit 022e15693390fae1a8e1eeb4c6c483c21e61cc2f.

grgia committed:
Use existing color for supporting text, change legal links to blue

grgia committed:
Remove unused variables, align footer with content

grgia committed:
Use only copyright as there is no login on this page

josemak25 commented:
I have read the CLA Document and I hereby sign the CLA

josemak25 commented:
> @josemak25 Can you please follow the instructions from the issue template and add a comment saying.
> 
> ```
> I have read the CLA Document and I hereby sign the CLA
> ```

Yes sorry my bad I have correct that I would fix this by removing this commit

maddylewis committed:
Update Your-Expensify-Partner-Manager.md

Removed the space so that the link shows correctly!

Commit:
update copy

Commit:
remove console log

Commit:
remove redundant call for deactivateLabel

tienifr committed:
fix: move getSortedReportActionsForDisplay to Utils and use in iouTransaction

bernhardoj committed:
move available iou options logic to util

bernhardoj committed:
show text based on iou options

bernhardoj committed:
add test

bernhardoj committed:
fix lint

bernhardoj committed:
add comment

s77rt committed:
Shift range end container while end offset is zero

bernhardoj committed:
fix lint again

mountiny commented on src/libs/EmojiUtils.js:
```diff
@@ -245,6 +245,15 @@ function suggestEmojis(text, limit = 5) {
+ * @param {String} text
+ * @returns {Array}
+ */
+function isEmoji(text) {
```
`contains` might be a better name for this

```suggestion
function containsEmoji(text) {
```

josemak25 replied:
Yea I would make this changes

ArekChr commented on src/styles/styles.js:
```diff
@@ -1100,6 +1101,19 @@ const styles = {
+        position: 'relative',
+        fontSize: variables.fontSizeEmoji,
+        lineHeight: variables.fontSizeEmojiHeight,
+        top: Platform.OS === 'android' ? 8 : 3,
```
I'm not sure if styles are different per platform, they should be splinted into separate folders like other styles, for android index.android.js, index.ios.js for ios and index.js for the web.
@mountiny, could you correct me if I'm wrong?

josemak25 replied:
Hi @parasharrajat relative to this comment
I was asked to separate the conditional styles into index.(ios | android | native).js which I am proposing is not so necessary for a small set of styles that is only applicable to emoji rendering, my reasons are creating files for both android, ios and web just to render emoji also increases  app size and for such a simple conditional style we can make way with Platform.select which is far better for this use case
I made a recent change on my last push to the open PR, which conditionally renders emoji. This is better than creating files just for a simple set of styles, what do you think?
 
 ```javascript
   emojiMessageText: {
        position: 'relative',
        fontSize: variables.fontSizeEmoji,
        lineHeight: variables.fontSizeEmojiHeight,
        ...Platform.select({
            ios: {
                top: 8,
            },
            android: {
                top: 8,
            },
            default: {
                top: 3,
            },
        }),
    },
 ```

Beamanator committed:
Little cleanup

Julesssss committed:
update Android icons to support all device versions

Julesssss committed:
put icon assets in default directory

Julesssss committed:
remove assets from debug flavour

Commit:
skip deactivateLabel invocation when autofill text exists

roryabraham commented on src/components/TextEmoji.js:
```diff
@@ -0,0 +1,55 @@
+
+    const propsStyle = StyleUtils.parseStyleAsArray(props.style);
+
+    return _.map([...Str.htmlDecode(props.children)], ((char, index) => (
```
So this component applies style to only the emoji characters in a string. So given a string like this:

```
smile U+1F600!
```

We would end up with a component hierarchy like this (ignoring keys):

```jsx
<>
    <Text>s</Text>
    <Text>m</Text>
    <Text>i</Text>
    <Text>l</Text>
    <Text>e</Text>
    <Text>{' '}</Text>
    <View>
        <Text style={propsStyle}>
            U+1F600
        </Text>
    </View>
    <Text>!</Text>
</>
```

But I think we could be smarter and land ourselves with a leaner component hierarchy like this:

```jsx
<>
    <Text>{'smile '}</Text>
    <View>
        <Text style={propsStyle}>
            U+1F600
        </Text>
    </View>
    <Text>!</Text>
</>
```

To do that, I think we could split the string using the emoji regex, rather than just mapping over all characters. The only thing is that we need to wrap the emoji regex in a capture group.

So given this input string:

```
smile U+1F600!
```

You'd want to create an `EmojiUtils` function called `splitStringByEmojis` that produces this as output:

```js
[
    {text: 'smile ', containsEmoji: false},
    {text: 'U+1F600', containsEmoji: true},
    {text: '!', containsEmoji: false},
]
```

Then in here you could loop over that array rather than the string.

roryabraham replied:
Also, I think it would be a good idea to write some unit tests to cover that function

parasharrajat reviewed (changes requested):
Please add comprehensive tests for each scenario from the issue https://github.com/Expensify/App/issues/4114#issue-946385786. 

If some of the scenarios are not implemented, please explain them why and also post the same on the issue as a comment.

parasharrajat reviewed (changes requested):
All commits must be GPG signed. Please do not squash the commits during the signing.

Commit:
check for the existence of platform specific method

marcaaron committed:
Fix avatar preview size

marcaaron committed:
support the svg avatars

Commit:
clear message so a refresh will allow for resending a code

fedirjh committed:
[Fix] Add missing Emoji.types

Commit:
add comment

marktoman committed:
Make line breaks HTML-compliant on native

fedirjh committed:
Fix lint

fedirjh committed:
[Fix] missing emojis.types in EmojiTest

marcochavezf committed:
pass the hubs data to javascript

marcochavezf committed:
check dynamically if the user is navigating bat to a hub page

bernhardoj committed:
merge with main

marktoman committed:
Merge package files

tienifr committed:
fix: toggle composer input only when the keyboard hide completely

tienifr committed:
fix: toggle composer input only when the keyboard hide completely

tienifr committed:
fix: toggle composer input only when the keyboard hide completely

tienifr committed:
fix: toggle composer input only when the keyboard hide completely

tienifr committed:
fix: create toggleReportActionComposeViewWhenCloseEditMessage function only be triggered when close edit message

tienifr committed:
fix: create toggleReportActionComposeViewWhenCloseEditMessage function only be triggered when close edit message

tienifr committed:
fix: issue

tienifr committed:
fix: delete redundant

tienifr committed:
fix: mweb bug

narefyev91 committed:
Keyboard is visible for a brief moment on request/send money page after getting back from currency search page

tienifr committed:
fix: add log again

tienifr committed:
fix: update returned value

tienifr committed:
Update src/libs/ReportActionsUtils.js

Co-authored-by: Luthfi <luthfi.ufi14@gmail.com>

narefyev91 committed:
Migrate event handler logic to Screen wrapper

narefyev91 committed:
Update src/components/ScreenWrapper/index.js

add spacing

Co-authored-by: Vit Horacek <36083550+mountiny@users.noreply.github.com>

Commit:
re-introduce formalertwrapper

fedirjh committed:
[Fix] Exclude policy rooms from getChatByParticipants results

roryabraham committed:
Sync LHN scroll with article scroll

fedirjh committed:
[Fix] race condition with updateAutomaticTimezone on connected devices

Commit:
update css selector for input elem that is autofilled

Commit:
don't try to select emoji when no emoji is highlighted

bernhardoj committed:
fix undefined personal details

eh2077 committed:
update expensify-common

bernhardoj committed:
fix iou hint text does not show on policy expense chat

Prince-Mendiratta committed:
conditionally polyfill Intl numberformat

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

situchan committed:
fix "Android - Keyboard is not dismissed after tapping outside of email input"

Prince-Mendiratta committed:
naming convention

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

s77rt committed:
blur active element before navigation and not after

alexxxwork committed:
Fixes expensify-common lib version for italic markdown fix

OSBotify committed:
Update version to 1.2.74-0

Commit:
fix lint issue

Commit:
Undo isloadingPayments change

neil-marcellini committed:
Throttle toggling test tools modal for mWeb Chrome

neil-marcellini committed:
Use constant for test tools modal throttle time

tienifr committed:
replace acceptedTerms with acceptTerms for consistency

tienifr committed:
update logic get default state for field

Prince-Mendiratta committed:
verbose naming

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

Prince-Mendiratta committed:
single function for common imports

Signed-off-by: Prince Mendiratta <prince.mendi@gmail.com>

narefyev91 committed:
Separate autofill logic between platforms

josemak25 commented:
> Please add comprehensive tests for each scenario from the issue [#4114 (comment)](https://github.com/Expensify/App/issues/4114#issue-946385786).
> 
> If some of the scenarios are not implemented, please explain them why and also post the same on the issue as a comment.

Test on all platforms

josemak25 commented:
> Please also sign your commits. I am traveling for the next 4 hours so I might not be able to test by that time.



> Please also sign your commits. I am traveling for the next 4 hours so I might not be able to test by that time.
The last commits to this push were signed, expect of cause I am missing something you can help point out thanks.

ntdiary committed:
replace onModalHide with onDismiss for the web

ntdiary committed:
refactor onDismiss

davidcardoza committed:
New US small business playbook

Mirroring a similar PR as https://github.com/Expensify/App/pull/14953

fedirjh committed:
Remove unused messages

zsgreenwald committed:
Update Smallbusinessplaybook

I removed the #Overview section at the top.

zsgreenwald committed:
Update Expensify-Playbook-for-US-based-VC-Backed-Startups.md

I ran the initial doc through Grammarly and made some revisions + fixed the issue with double quoting the quotes in this article.

zsgreenwald committed:
Create US-Based-Bootstrapped-Startup Playbook

This is a new playbook to be added to the Playbook Hub in help.expensify.com

zsgreenwald committed:
Update Expensify-Playbook for US-based VC-backed startups.md

Changed the naming of Continuous Reconciliation to Auto-Reconciliation

allroundexperts committed:
fix: updated dob check logic

ArekChr commented on src/styles/styles/styles.native.js:
```diff
@@ -1,24 +1,24 @@
-import fontFamily from './fontFamily';
```
I'm not sure about a change to this file. If some component requires different styles per platform, there should be a separate folder for these specific styles, not splitting global styles

josemak25 replied:
The `styles.native.js` file was just the base existing `styles.js` file I renamed it due to the folder and platform splitting structure there.
Not sure I understand `there should be a separate folder for these specific styles, not splitting global styles` please can you explain more @ArekChr

ArekChr replied:
If styles are different per specific platform, then a new file for those styles should be created instead of splitting global styles. e.g. styles/emojiText/styles.native.js.
@parasharrajat, could you correct me if I'm wrong?

ArekChr commented on src/styles/styles/index.js:
```diff
@@ -0,0 +1,9 @@
+import {Platform} from 'react-native';
+
+const styles = Platform.select({
+    ios: require('./styles.ios').default,
+    default: require('./styles.native').default,
+    android: require('./styles.android').default,
+});
+
+export default styles;
```
This file is not needed

josemak25 replied:
@ArekChr it's actually needed cause when I removed it, the platform-specific styles are not picked, maybe I am missing something here, mind pointing me on how to fix this thanks 🙏

ArekChr replied:
It should work if you remove styles.native

ArekChr replied:
There might be conflict between ios/android and native file

josemak25 replied:
Okay so the way styles are imported on the app is via `[path]/styles/styles.js`
So removing index.js files means I have to update `styles.native.js` to `index.js` and `styles.ios.js` to index.ios.js and lastley `styles.android.js` to index.android.js

And since iOS and Android still need to extend the global styles we have to import the index.js into iOS and Android which would lead to Require cycles and throw errors.

<img width="326" alt="Screenshot 2023-02-22 at 10 32 26 AM" src="https://user-images.githubusercontent.com/33870502/220579759-83ef00ac-3432-40f8-92e8-a73811c81455.png">
<img width="564" alt="Screenshot 2023-02-22 at 10 35 52 AM" src="https://user-images.githubusercontent.com/33870502/220580560-de08cb99-5d4a-4f23-8a54-8795aaf9eba5.png">

ArekChr replied:
Require cycle makes styles undefined, please fix that

Gonals committed:
Do not add markdown if the comment is too long

Gonals committed:
also skip conversion down the line

Gonals committed:
add comments

tienifr committed:
update method's description

tienifr committed:
fix: the comment of getSortedReportActionsForDisplay function

ntdiary committed:
need to consider coverScreen

ntdiary committed:
Remove unused code

ntdiary committed:
Remove unused code

roryabraham committed:
Prevent LHN scroll from causing jank

puneetlath committed:
dont send policy members to the api if they have errors

puneetlath committed:
update comment for new logic

alexxxwork committed:
Adds package-lock.json

Commit:
update comment

roryabraham commented on src/libs/EmojiUtils.js:
```diff
@@ -245,6 +245,66 @@ function suggestEmojis(text, limit = 5) {
+ * @param {String} text
+ * @returns {Array}
+ */
+function containsEmoji(text) {
```
This is no longer an accurate function name.

josemak25 replied:
I changed this to `getAllEmojiFromText` what do you think?

OSBotify committed:
Update version to 1.2.75-0

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

davidcardoza committed:
Update SmallbusinessPlaybook

Co-authored-by: Scott Deeter <deetergp@users.noreply.github.com>

roryabraham committed:
Revert "Don't show empty chat"

marcochavezf committed:
remove empty spaces

marcochavezf committed:
disable rule because we're not using underscore in expensify help

marcochavezf committed:
add .md extension

marcochavezf committed:
remove empty spaces

marcochavezf committed:
rename file to match title

marcochavezf committed:
add new entry to routes

marcochavezf committed:
update formatting

s77rt committed:
Updated comment

josemak25 committed:
add lint fix and change EmojiUtils.containsEmoji to EmojiUtils.getAllEmojiFromText

narefyev91 committed:
Add prop type for autoComplete

narefyev91 committed:
autoComplete must be required

narefyev91 committed:
Date format placeholder is missing for Date field on the date of birth page

Julesssss committed:
fix sms email suffix on sign in page

narefyev91 committed:
Add prop to use navigation handler for control keyboard visibility

tienifr committed:
fix: remove filterReportActionsForDisplay function and update UTs

Julesssss committed:
simplify login format solution and align with the go back message

iwiznia committed:
Fix tests

marktoman committed:
Make auto-emails work on multi-line

rushatgabhane committed:
fix space

OSBotify committed:
Update version to 1.2.76-0

OSBotify committed:
Update version to 1.2.76-1

OSBotify committed:
Update version to 1.2.76-2

bernhardoj committed:
allow option row to have multiple lines of alternate text

Julesssss committed:
increase JVM memory allowance

Julesssss committed:
use exactly double the JVM mem allowance

bernhardoj committed:
add comment

Gonals committed:
add to CONST

Gonals committed:
One more place

dylanexpensify committed:
Create TestRail_Navigation_C+.md

Create a new page for C+ learning how to navigate and complete TestRail regression test step updates/creation.

dylanexpensify committed:
Update TestRail_Navigation_C+.md

OSBotify committed:
Update version to 1.2.76-3

dylanexpensify committed:
Rename TestRail_Navigation_C+.md to contributingGuides/TestRail_Navigation_C+.md

dylanexpensify committed:
Update TestRail_Navigation_C+.md

dylanexpensify committed:
Update TestRail_Navigation_C+.md

allroundexperts committed:
feat: added withoutFocusOnSecondaryInteraction prop on PressableWithSecondaryIntegration component

situchan committed:
close modal when opening new page using keyboard shortcut

neil-marcellini committed:
Revert "Open a test tools modal with 5 taps on dev"

OSBotify committed:
Update version to 1.2.76-4

situchan committed:
add explanation of setCloseModal calls

marktoman committed:
Update expensify-common

marktoman committed:
Fix casing

OSBotify committed:
Update version to 1.2.76-5

OSBotify committed:
Update version to 1.2.76-6

OSBotify committed:
Update version to 1.2.76-7

ntdiary committed:
rename useOnDismiss to shouldUseOnDismiss

bernhardoj committed:
fix infinite loading

dylanexpensify committed:
Update contributingGuides/TestRail_Navigation_C+.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

dylanexpensify committed:
Update contributingGuides/TestRail_Navigation_C+.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

dylanexpensify committed:
Update contributingGuides/TestRail_Navigation_C+.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

dylanexpensify committed:
Update contributingGuides/TestRail_Navigation_C+.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

dylanexpensify committed:
Update contributingGuides/TestRail_Navigation_C+.md

Co-authored-by: Alex Beaman <alexbeaman@expensify.com>

situchan committed:
update comment

Co-authored-by: Puneet Lath <puneet@expensify.com>

situchan committed:
update comment

bondydaa committed:
Revert "replace onModalHide with onDismiss for the web"

bondydaa committed:
Revert "Revert "replace onModalHide with onDismiss for the web""

josemak25 committed:
add check for empty message on hasEmojis function

roryabraham commented:
Yeah @josemak25 something is super borked here. Let's just close this and open a fresh PR w/ all commits signed please.

mountiny commented:
Agreed with Rory, that might be easier

josemak25 commented:
> Yeah @josemak25 something is super borked here. Let's just close this and open a fresh PR w/ all commits signed please.

Yea tried rebasing and got this crazy commit history
Already on that, a new PR should be up in less than 20m

josemak25 commented:
> Agreed with Rory, that might be easier

@roryabraham yea can't agree more, I would do so in less than 20m

josemak25 commented:
@roryabraham @mountiny a new PR has been raised for this as this PR is to be closed here is the new link https://github.com/Expensify/App/pull/15516
